src/intrinsics/prepack/global.js

```diff
@@ -227,14 +227,19 @@ export default function(realm: Realm): void {
       "__evaluatePureFunction",
       0,
       (context, [functionValue]) => {
-        invariant(functionValue instanceof ECMAScriptSourceFunctionValue);
-        invariant(typeof functionValue.$Call === "function");
-        let functionCall: Function = functionValue.$Call;
-        return realm.evaluatePure(
-          () => functionCall(realm.intrinsics.undefined, []),
-          /*bubbles*/ true,
-          /*reportSideEffectFunc*/ null
-        );
+        if (functionValue instanceof ECMAScriptSourceFunctionValue && typeof functionValue.$Call === "function") {
+          let functionCall: Function = functionValue.$Call;
+          return realm.evaluatePure(
+            () => functionCall(realm.intrinsics.undefined, []),
+            /*bubbles*/ true,
+            /*reportSideEffectFunc*/ null
+          );
+        } else {
+          realm.handleError(
+            new CompilerDiagnostic(`CompilerDiagnostic Error`, realm.currentLocation, "PP1005", "FatalError")
```

Comment:
We need an error message that will help a developer pin point what is wrong and help them understand how to fix it. The error code needs to be unique and there should be a wiki page that documents the error in some detail, preferably along with example code.